product_brand_tag_secondary/tests/test_product_brand.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2021 Camptocamp SA
+# @author: Simone Orsi <simone.orsi@camptocamp.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from psycopg2 import IntegrityError
+
+from odoo import exceptions
+from odoo.tools import mute_logger
+
+from odoo.addons.product_brand.tests.common import CommonCase
+
+
+class TestBrandTags(CommonCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.product_brand2 = cls.product_brand.copy({"name": "Test Brand 2"})
+        cls.product_brand3 = cls.product_brand.copy({"name": "Test Brand 3"})
+        cls.tag = cls.env["product.brand.tag"].create(
+            {
+                "name": "Test Tag",
+                "product_brand_ids": [(6, 0, [cls.product_brand3.id])],
+                "secondary_product_brand_ids": [
+                    (6, 0, [cls.product_brand.id, cls.product_brand2.id])
+                ],
+            }
+        )
+
+    def test_brand_rel(self):
+        self.assertEqual(self.product_brand.secondary_tag_ids, self.tag)
+        self.assertEqual(self.product_brand2.secondary_tag_ids, self.tag)
+        self.assertEqual(self.product_brand3.tag_ids, self.tag)
+
+    def test_brand_rel_check(self):
+        with self.assertRaisesRegex(
+            exceptions.UserError, self.tag._check_brands_relation_error_msg(self.tag)
```

Comment:
I would prefer to have imported `UserError` instead of `exceptions`, and then use directly it.
But it's not blocking.

Reply:
I like it this way, sorry :stuck_out_tongue_winking_eye: